### PR TITLE
The assistant attempts to scavenge while dead. #131

### PR DESCRIPTION
**Pull request by d0tsd0tsd0ts**
Changelog:
 * Included Player is Dead. 
 * Fix ignore item after scavenger 1x
 * FIx double scavenger for 1 item
 * Removed CheckItemContainer it's not updating bcs lock section

### CODE REVIEW

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 1]
```diff
@@ -183,10 +191,12 @@ private void ItemsOnCollectionChanged( int totalcount, bool added, Item[] items
 
         internal void CheckArea()
         {
-            if ( !Enabled || Engine.Player == null )
+            //BEGIN
+            if (!Enabled || Engine.Player == null || Engine.Player.IsDead)
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:195]: Fix the formatting... If ( !Enabled ) spaces between parentheses
Remove comments

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 2]
```diff
@@ -14,6 +14,14 @@
 using Newtonsoft.Json.Linq;
 using UOC = ClassicAssist.UO.Commands;
 
+/*
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:17]: Remove comments

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 3]
```diff
@@ -183,10 +191,12 @@ private void ItemsOnCollectionChanged( int totalcount, bool added, Item[] items
 
         internal void CheckArea()
         {
-            if ( !Enabled || Engine.Player == null )
+            //BEGIN
+            if (!Enabled || Engine.Player == null || Engine.Player.IsDead)
             {
                 return;
             }
+            //END
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:199]: Remove

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 4]
```diff
@@ -216,6 +226,10 @@ internal void CheckArea()
                     scavengerItems.AddRange( matches );
                 }
 
+                //BEGIN
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:229]: Remove comments

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 5]
```diff
@@ -228,20 +242,19 @@ internal void CheckArea()
                     return;
                 }
 
-                foreach ( Item scavengerItem in scavengerItems.Where( scavengerItem =>
-                    scavengerItem.Distance <= SCAVENGER_DISTANCE ).Distinct() )
+                //BEGIN
+                foreach (Item scavengerItem in scavengerItems.Where(i =>
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:246]: Fix spaces between parentheses

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 6]
```diff
@@ -228,20 +242,19 @@ internal void CheckArea()
...
+                  i.Distance <= SCAVENGER_DISTANCE && !_ignoreList.Contains(i.Serial)))
                 {
-                    UOC.SystemMessage( string.Format( Strings.Scavenging___0__, scavengerItem.Name ?? "Unknown" ),
-                        (int) UOC.SystemMessageHues.Yellow );
-                    Task<bool> t = ActionPacketQueue.EnqueueDragDrop( scavengerItem.Serial, scavengerItem.Count,
+                    UOC.SystemMessage(string.Format(Strings.Scavenging___0__, scavengerItem.Name ?? "Unknown"), 61);
+                    Task<bool> t = ActionPacketQueue.EnqueueDragDrop(scavengerItem.Serial, scavengerItem.Count,
                         container.Serial, QueuePriority.Low, true, true, requeueOnFailure: false,
-                        successPredicate: CheckItemContainer );
+                        successPredicate: CheckItemContainer);
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:252]: Fix spaces between parentheses

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 7]
```diff
@@ -228,20 +242,19 @@ internal void CheckArea()
...
 
-                    if ( t.Result && CheckItemContainer( scavengerItem.Serial, container.Serial ) )
-                    {
-                        _ignoreList.Add( scavengerItem.Serial );
-                    }
+                    if (t.Result)
+                        _ignoreList.Add(scavengerItem.Serial);
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:255]: Add braces even for single line if

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs [hunk 8]
```diff
@@ -228,20 +242,19 @@ internal void CheckArea()
...
                 }
+                //END
```
Comment (Reetus) [inline ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs:257]: Remove comment

**Review by Reetus — CHANGES_REQUESTED** [review]

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 2 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 1 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 3 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 4 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 5 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 6 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 8 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

**Review by d0tsd0tsd0ts — COMMENTED** [review]
Requests done

File: ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs (hunk 7 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UI/ViewModels/Agents/ScavengerTabViewModel.cs]: Done

**Pull request merged**